toolchain/check/impl_lookup.cpp

```diff
@@ -373,29 +376,29 @@ auto LookupImplWitness(Context& context, SemIR::LocId loc_id,
 
   if (FindAndDiagnoseImplLookupCycle(context, context.impl_lookup_stack(),
                                      loc_id, type_const_id,
-                                     interface_const_id)) {
-    return SemIR::ErrorInst::SingletonInstId;
+                                     query_facet_type_const_id)) {
+    return SemIR::InstBlockIdOrError::MakeError();
   }
 
   bool has_other_requirements = false;
   auto interfaces = GetInterfacesFromConstantId(
-      context, loc_id, interface_const_id, has_other_requirements);
+      context, loc_id, query_facet_type_const_id, has_other_requirements);
   if (interfaces.empty()) {
     // TODO: Remove this when the context.TODO() is removed in
     // GetInterfacesFromConstantId.
-    return SemIR::InstId::None;
+    return SemIR::InstBlockId::None;
   }
   if (has_other_requirements) {
     // TODO: Remove this when other requirements go away.
-    return SemIR::InstId::None;
+    return SemIR::InstBlockId::None;
   }
```

Comment:
(This could be done in a follow-up, but this PR opens the door to fixing the issue.)

Also delete the `context.TODO(`...`)` call in `GetInterfacesFromConstantId`

```suggestion
  if (has_other_requirements) {
    // TODO: Remove this when other requirements go away.
    return SemIR::InstBlockId::None;
  }
  if (interfaces.empty()) {
    return SemIR::InstBlockId::Empty;
  }
```

Reply:
Done